Frends.RabbitMQ.Publish/Frends.RabbitMQ.Publish.Tests/QuorumQueueTests.cs

```diff
@@ -0,0 +1,350 @@
+{
+    /// <summary>
+    /// You will need access to RabbitMQ queue, you can create it e.g. by running
+    /// docker run -d --hostname my-rabbit -p 5672:5672 -p 8080:1567 -p 15672:15672 -e RABBITMQ_DEFAULT_USER=agent -e RABBITMQ_DEFAULT_PASS=agent123 rabbitmq:3.9-management
+    /// In that case URI would be amqp://agent:agent123@localhost:5672 
+    /// Access UI from http://localhost:15672 username: agent, password: agent123
+    /// </summary>
+
+    private const string TestUri = "amqp://agent:agent123@localhost:5672";
+    private const string TestHost = "localhost";
+    private const string queue = "quorum";
+    private const string exchange = "exchange";
+
+    [TestInitialize]
+    public void CreateExchangeAndQueue()
+    {
+        var factory = new ConnectionFactory { Uri = new Uri(TestUri) };
+        using var connection = factory.CreateConnection();
+        using var channel = connection.CreateModel();
+        channel.ExchangeDeclare(exchange, type: "fanout", durable: false, autoDelete: false);
+        var args = new Dictionary<string, object>();
+        args["x-queue-type"] = "quorum";
+        channel.QueueDeclare(queue, durable: true, exclusive: false, autoDelete: false, arguments: args);
+        channel.QueueBind(queue, exchange, routingKey: "");
+    }
+
+    [TestCleanup]
+    public void DeleteExchangeAndQueue()
+    {
+        Helper.DeleteQuorumQueue(TestUri, queue, exchange);
+    }
+
+    [TestMethod]
+    public void TestPublishAsString()
+    {
+        Connection connection = new()
+        {
+            Host = TestHost,
+            Username = "agent",
+            Password = "agent123",
+            RoutingKey = "quorum",
+            QueueName = "quorum",
+            Create = false,
+            Durable = false,
+            AutoDelete = false,
+            AuthenticationMethod = AuthenticationMethod.Host,
+            ExchangeName = ""
+        };
+
+        Input input = new()
+        {
+            DataString = "test message",
+            InputType = InputType.String,
+
+            Headers = new Header[] {
+            new Header { Name = "X-AppId", Value = "application id" },
+            new Header { Name = "X-ClusterId", Value = "cluster id" },
+            new Header { Name = "Content-Type", Value = "content type" },
+            new Header { Name = "Content-Encoding", Value = "content encoding" },
+            new Header { Name = "X-CorrelationId", Value = "correlation id" },
+            new Header { Name = "X-Expiration", Value = "100" },
+            new Header { Name = "X-MessageId", Value = "message id" },
+            new Header { Name = "Custom-Header", Value = "custom header" }
+        }
+        };
+
+        var readValues = new Helper.ReadValues();
+        var result = RabbitMQ.Publish(input, connection);
+        Helper.ReadMessage(readValues, connection);
+
+        Assert.IsTrue(!string.IsNullOrEmpty(readValues.Message) && readValues.Message.Equals("test message"));

```

Comment:
Please move these checks to their own asserts and use `AreEqual` for equal comparisons.